CRYOSMSApp/src/CRYOSMSDriver.cpp

```diff
@@ -1275,11 +1278,13 @@ extern "C"
 		std::string devPrefix = args[0].aval.av[2];
 
 		errlogSevPrintf(errlogInfo, "Loading macros into asyn driver:\n");
-		for (int i = 0; i < sizeof(envVarsNames) / sizeof(const char*); ++i)
+		for (std::size_t i = 0; i < sizeof(envVarsNames) / sizeof(const char*); ++i)
 		{
-			char * argval = args[0].aval.av[i + 3];
-			errlogSevPrintf(errlogInfo, "%s: %s\n", envVarsNames[i], argval);
-			std::pair<std::string, std::string > newPair(envVarsNames[i], argval); // args starts with CRYOSMSConfigure, portName and devPrefix
+			char * argValC = args[0].aval.av[i + 3];
+			std::string argVal = argValC;
+			std::string argName = envVarsNames[i];
+			errlogSevPrintf(errlogInfo, "%s: %s\n", envVarsNames[i], argValC);
```

Comment:
Could you avoid argValC with
```
std::string argVal(args[0].aval.av[i + 3])
```
and then use `argVal.c_str()` insread of `argValC`